src/main/java/pt/tqs/g205/security/JwtAuthorizationFilter.java

```diff
@@ -0,0 +1,59 @@
+package pt.tqs.g205.security;
+
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.web.authentication.www.BasicAuthenticationFilter;
+
+import java.io.IOException;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class JwtAuthorizationFilter extends BasicAuthenticationFilter {
+  private JwtUtil jwtUtil;
+  private UserDetailsService userDetailsService;
+
+  /**
+   * Constructor.
+   * @param authenticationManager authentication manager.
+   * @param jwtUtil instancia de JwtUtil.
+   * @param userDetailsService instancia de UserDetailsService.
+   */
+  public JwtAuthorizationFilter(AuthenticationManager authenticationManager, JwtUtil jwtUtil,
+      UserDetailsService userDetailsService) {
+    super(authenticationManager);
+    this.jwtUtil = jwtUtil;
+    this.userDetailsService = userDetailsService;
+  }
+
+  @Override
+  protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response,
+      FilterChain chain) throws IOException, ServletException {
+    String header = request.getHeader("Authorization");
+    if (header != null && header.startsWith("Bearer ")) {
+      UsernamePasswordAuthenticationToken auth = getAuthentication(request, header.substring(7));
+      if (auth != null) {
+        SecurityContextHolder.getContext().setAuthentication(auth);
+      }
+    }
+    chain.doFilter(request, response);
+  }
+
+  private UsernamePasswordAuthenticationToken getAuthentication(HttpServletRequest request,
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unused method parameter "request". ([squid:S1172](https://sonarcloud.io/organizations/bdomingues-github/rules#rule_key=squid%3AS1172))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=pt.tqs.g205%3Arestapi&pullRequest=14&issues=AWOs6QOC_Tl7jhO82Lfm&open=AWOs6QOC_Tl7jhO82Lfm)

[ISSUE_KEY]: AWOs6QOC_Tl7jhO82Lfm